src/lib.rs

```diff
@@ -436,11 +459,11 @@ impl Compressor {
 
         // SAFETY: `end` will point just after the end of the `plaintext` slice.
         let in_end = unsafe { in_ptr.byte_add(plaintext.len()) };
-        let in_end_sub8 = unsafe { in_end.byte_sub(8) };
+        let in_end_sub8 = in_end as usize - 8;
```

Comment:
miri caught that the old thing was technically a dangling ptr.

i never dereferenced it but just to make it happy i work with it as an address instead of as a pointer